src/TeamStore.DevUI/Controllers/HomeController.cs

```diff
@@ -30,6 +34,27 @@ public HomeController(IProjectsService projectsService)
             return View(homeViewModel);
         }
 
+        [HttpGet]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> GetAssetResults()
+        {
+            string accessIpAddress = HttpContext?.Connection?.RemoteIpAddress?.ToString();
+            if (string.IsNullOrWhiteSpace(accessIpAddress)) return BadRequest();
+
+            try
+            {
+                var assets = await _assetService.GetAssetResultsAsync();
```

Comment:
The big problem here is exposing service-layer objects to HTTP results. This should collect a List<Asset> and run through a Factory before returning to the UI, just like the Projects controller. There are several actions left that still return a "Project" entity instead of a "ProjectViewModel", but they are wrong.